lib/index.ts

```diff
@@ -23,21 +37,44 @@ export async function buildGoPkgDepTree(
 // in particular, it cannot build the proper dependency graph (only a flat list).
 export async function buildGoVendorDepTree(
   manifestFileContents: string,
-  options?: any): Promise<DepTree> {
+  options?: unknown): Promise<DepTree> {
   return buildGoDepTree(parseGoVendorConfig(manifestFileContents));
 }
 
+// We are not using go.sum file here because it's not actually a lockfile and contains dependencies
+// that are actually long gone.
+export function buildGoModDepTree(
+  manifestFileContents: string,
+  options?: unknown,
+): DepTree {
+  // We actually use only some bits of the go.mod contents
+  const goMod = parseGoMod(manifestFileContents);
+  const depTree: DepTree = {
+    name: goMod.moduleName,
+    version: '0.0.0',
+    dependencies: {},
+  };
+  const dependencies = depTree.dependencies!;
+  for (const req of goMod.requires) {
+    dependencies[req.moduleName] = {
+      name: req.moduleName,
+      version: toSnykVersion(req.version),
```

Comment:
why don't we just do this conversion in the gomod parser?

Reply:
Because Lili wanted to "preserve 100% of the information". The idea for GoMod is to be complete and lossless representation of the original file.